projects/fsastorefrontlib/src/cms-components/checkout/guards/add-options-guard.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable } from '@angular/core';
+import {
+  ActivatedRoute,
+  ActivatedRouteSnapshot,
+  CanActivate,
+  RouterStateSnapshot,
+  UrlTree,
+} from '@angular/router';
+import { Observable } from 'rxjs';
+import { FSCartService } from '../../../core';
+import { map } from 'rxjs/operators';
+import {
+  MultiCartService,
+  OCC_USER_ID_ANONYMOUS,
+  UserIdService,
+} from '@spartacus/core';
+import { Cart } from '@spartacus/core';
+@Injectable({
+  providedIn: 'root',
+})
+export class AddOptionsGuard implements CanActivate {
+  constructor(
+    protected fsCartService: FSCartService,
+    protected activatedRoute: ActivatedRoute,
+    protected userIdService: UserIdService,
+    protected multiCartService: MultiCartService
+  ) {}
+
+  newCart$: Observable<Cart>;
+
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    _: RouterStateSnapshot
+  ): Observable<boolean | UrlTree> {
+    return this.userIdService.getUserId().pipe(
+      map(userId => {
+        let guid = route.queryParams['guid'];
+        this.setActiveCartForUser(guid, userId);
+        return true;
+      })
+    );
+  }
+
+  setActiveCartForUser(guid: string, userId: string) {
+    if (guid) {
```

Comment:
Maybe check this guid in `canActivate` and return false if it doesn't exist? I'm also wondering is it possible it doesn't exist? Maybe this check is not needed at all then.